src/lib.rs

```
@@ -9,59 +9,198 @@
+            netlink_tc::Nla::Fcnt(bytes) => attrs.push(TcAttr::Fcnt(bytes)),
+            netlink_tc::Nla::Stab(bytes) => attrs.push(TcAttr::Stab(bytes)),
+            netlink_tc::Nla::Chain(bytes) => attrs.push(TcAttr::Chain(bytes)),
+            netlink_tc::Nla::HwOffload(byte) => attrs.push(TcAttr::HwOffload(byte)),
+            _ => (),
+        }
+    }
+
+    TcMsg { header, attrs }
 }
 
-/// Get list of all `tc` qdiscs.
-pub fn qdiscs<T: netlink::NetlinkConnection>() -> Result<Vec<Tc>, errors::TcError> {
-    tc::qdiscs::<T>()
+fn to_link(link_message: NlLinkMessage) -> LinkMsg {
+    let NlLinkMessage {
+        header: link_header,
+        nlas,
+        ..
+    } = link_message;
+    let header = LinkHeader {
+        index: link_header.index,
+    };
+
+    let mut name = String::new();
+    for nla in nlas {
+        if let netlink_link::nlas::Nla::IfName(if_name) = nla {
+            name = if_name;
+        }
+    }
+
+    LinkMsg {
+        header,
+        attr: LinkAttr { name },
+    }
 }
 
-/// Get list of all `tc` classes.
-pub fn classes<T: netlink::NetlinkConnection>() -> Result<Vec<Tc>, errors::TcError> {
-    tc::classes::<T>()
+fn parse(
+    messages: Vec<NetlinkMessage<RtnlMessage>>,
+) -> Result<Vec<RtNetlinkMessage>, NetlinkError> {
+    let mut tc_messages = Vec::new();
+    for message in messages {
+        match message.payload {
+            NetlinkPayload::InnerMessage(RtnlMessage::NewQueueDiscipline(message)) => {
+                tc_messages.push(RtNetlinkMessage::GetQdisc(to_tc(message.clone())))
+            }
+            NetlinkPayload::InnerMessage(RtnlMessage::NewTrafficClass(message)) => {
+                tc_messages.push(RtNetlinkMessage::GetClass(to_tc(message.clone())))
+            }
+            NetlinkPayload::InnerMessage(RtnlMessage::NewLink(message)) => {
+                tc_messages.push(RtNetlinkMessage::GetLink(to_link(message.clone())))
+            }
+            _ => (),
```

Is silently ignoring the right thing to do here? Might be better to let caller choose. Similar to std library `OpenOptions`, could have a builder for parse options. One option could be to ignore unknown; otherwise fail